src/sync/impl/sync_file.hpp

```diff
@@ -0,0 +1,91 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_SYNC_FILE_HPP
+#define REALM_OS_SYNC_FILE_HPP
+
+#include <string>
+
+namespace realm {
+
+namespace util {
+
+enum class FilePathType {
+    File, Directory
+};
+
+/// Given a string, turn it into a percent-encoded string.
+std::string make_percent_encoded_string(const std::string& raw_string);
```

Comment:
I suspect all of these could take `realm::StringData` as arguments rather than `std::string`.